pkg/controllers/management/auth/token_test.go

```diff
@@ -0,0 +1,373 @@
+package auth
+
+import (
+	"fmt"
+	"strings"
+	"testing"
+	"time"
+
+	v3 "github.com/rancher/rancher/pkg/apis/management.cattle.io/v3"
+	tokens2 "github.com/rancher/rancher/pkg/auth/tokens"
+	"github.com/rancher/rancher/pkg/features"
+	"github.com/rancher/rancher/pkg/generated/norman/management.cattle.io/v3/fakes"
+	"github.com/stretchr/testify/assert"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+)
+
+type tokenTestCase struct {
+	inputToken                  *v3.Token
+	expectedOutputToken         *v3.Token
+	inputUserAttribute          *v3.UserAttribute
+	expectedOutputUserAttribute *v3.UserAttribute
+	enableHashing               bool
+	description                 string
```

Comment:
I don't see this description being used directly by the test, could it be omitted? The description could be added as a comment next to the test case.

Reply:
I instead added an error message when a test case fails using the description. Is that okay?

Reply:
:+1: